seed/views/v3/properties.py

```diff
@@ -1010,6 +1014,244 @@ def update(self, request, pk=None):
+        in views/properties.py because if this is a buildingsync update, then the buildingsync file may
+        contain new or removed items that we want to merge.
+
+        :param old_state: PropertyState
+        :param new_state: PropertyState
+        :return: PropertyState, updated new_state
+        """
+        # for s in old_state.scenarios.all():
+        #     s.property_state = new_state
+        #     s.save()
+        #
+        # # Move the measures to the new state
+        # for m in PropertyMeasure.objects.filter(property_state=old_state):
+        #     m.property_state = new_state
+        #     m.save()
+        #
+        # # Move the old building file to the new state to preserve the history
+        # for b in old_state.building_files.all():
+        #     b.property_state = new_state
+        #     b.save()
+        #
+        # for s in Simulation.objects.filter(property_state=old_state):
+        #     s.property_state = new_state
+        #     s.save()
+
+        return new_state
+
+    @swagger_auto_schema(
+        manual_parameters=[
+            AutoSchemaHelper.path_id_field(
+                description='ID of the property (not property view) to update'
+            ),
+            AutoSchemaHelper.query_org_id_field(),
+            AutoSchemaHelper.query_integer_field(
+                'cycle_id',
+                required=True,
+                description='ID of the cycle to '
+            ),
+            AutoSchemaHelper.upload_file_field(
+                'file',
+                required=True,
+                description='BuildingSync file to use',
+            ),
+            AutoSchemaHelper.form_string_field(
+                'file_type',
+                required=True,
+                description='Either "Unknown" or "BuildingSync"',
+            ),
+        ],
+        request_body=no_body,
+    )
+    @action(detail=True, methods=['PUT'], parser_classes=(MultiPartParser,))
+    @has_perm_class('can_modify_data')
+    def update_with_building_sync(self, request, pk):
+        """
+        Update an existing PropertyView with a building file. Currently only supports BuildingSync.
+        """
+        if len(request.FILES) == 0:
+            return JsonResponse({
+                'success': False,
+                'message': "Must pass file in as a Multipart/Form post"
+            })
+
+        the_file = request.data['file']
+        file_type = BuildingFile.str_to_file_type(request.data.get('file_type', 'Unknown'))
+        organization_id = request.query_params.get('organization_id', None)
+        cycle_pk = request.query_params.get('cycle_id', None)
+
+        if not cycle_pk:
+            return JsonResponse({
+                'success': False,
+                'message': "Cycle ID is not defined"
+            })
+        else:
+            cycle = Cycle.objects.get(pk=cycle_pk)
+
+        result = self._get_property_view_for_property(pk, cycle_pk)
```

Comment:
I'd like to refactor this view to just accept a pk that's a property_view, not a canonical property id. Then we're much more consistent and we don't have to fool with this method. Thoughts @adrian-lara ?

Reply:
Agree - that's a good path to take.